pkg/repo/chartrepo.go

```diff
@@ -258,6 +322,68 @@ func FindChartInAuthRepoURL(repoURL, username, password, chartName, chartVersion
 	return absoluteChartURL, nil
 }
 
+// FindChartInRepoURLWithAuth finds chart in chart repository pointed by repoURL
+// without adding repo to repositories, like FindChartInRepoURL,
+// but it also receives credentials for the chart repository.
+func FindChartInRepoURLWithAuth(getters getter.Providers, opts ...Option) (string, error) {
+
+	// Download and write the index file to a temporary location
+	buf := make([]byte, 20)
+	rand.Read(buf)
+	name := strings.ReplaceAll(base64.StdEncoding.EncodeToString(buf), "/", "-")
+	repoOpts := options{}
+	for _, opt := range opts {
+		opt(&repoOpts)
+	}
+
+	c := Entry{
+		URL:      repoOpts.repoURL,
+		Username: repoOpts.username,
+		Password: repoOpts.password,
+		CertFile: repoOpts.certFile,
+		KeyFile:  repoOpts.keyFile,
+		CAFile:   repoOpts.caFile,
+		Name:     name,
+		Token:    repoOpts.token,
+	}
+	r, err := NewChartRepository(&c, getters)
```

Comment:
Since most of this (besides relating to token) is reused from FindChartInAuthRepoURL, can we move it into a private helper function such as `findChartInAuthRepoURL(token)` so that the file has overall less lines of code?

Reply:
I have updated my commit, would you please review again? Thanks.